singer_sdk/streams/core.py

```diff
@@ -1153,9 +1175,7 @@ def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
     # Abstract Methods
 
     @abc.abstractmethod
-    def get_records(
-        self, context: Optional[dict]
-    ) -> Iterable[Union[dict, Tuple[dict, dict]]]:
+    def get_records(self, context: Optional[dict]) -> Iterable[dict]:
```

Comment:
Can you say more about the change in the return signature?

Admittedly this is an awkward design, but for some parent child use cases it may still be in use, per the commend in the below docstring:

> Parent streams can optionally return a tuple, in which case the second item in the tuple being a `child_context` dictionary for the stream's `context`.

Primarily this is used when the child context is some python object that should not be part of the record - such as a python Tableau workbook object in the `tap-tableau` case.

Reply:
@aaronsteers No you're right. I've restored the signature with the optional `tuple` return value. It was an initial oversight on my part cause I thought the `child_context` returned from `get_records` was never used but it is.

We do seem to be using `copy.copy` a few times very defensively in case developers want (try?) to mutate the context dict.

Reply:
@edgarrmondragon - 

> We do seem to be using copy.copy a few times very defensively in case developers want (try?) to mutate the context dict.

I wish there was a convenient class for read-only `dict`. We often want to send contexts (and config) around between SDK code and dev-written code, and yes, the key word is 'defensive'. If a dev mutates or alters these, there could be adverse affects.

That said, I'm not going to hold tightly to those `copy` operators if you think we can remove them, I'll be okay with that.